packages/react-compose/src/createClassResolver.test.ts

```diff
@@ -0,0 +1,47 @@
+import { createClassResolver } from './createClassResolver';
+
+describe('createClassResolver', () => {
+  const classResolver = createClassResolver({
+    root: 'root',
+    slot1: 'slot1',
+    slot2: 'slot2',
+    primary: 'primary',
+    size_small: 'small',
+    size_medium: 'medium',
+  });
+  const slots = { root: null, slot1: null, slot2: null };
+
+  it('can resolve slot classes', () => {
+    expect(classResolver({}, slots)).toEqual({
+      root: 'root',
+      slot1: 'slot1',
+      slot2: 'slot2',
+    });
+  });
+
+  it('can resolve modifiers', () => {
+    expect(classResolver({ primary: true }, slots)).toEqual({
```

Comment:
with our scss conversions, we are actually adding modifier classes onto every slot. that way the scss can be written in a cleaner way.